app/controllers/comments_controller.rb

```diff
@@ -1,32 +1,52 @@
 class CommentsController < ApplicationController
+  before_action :set_post, only: %i[index create show update destroy]
+
+  def index
+    json_response(@post.comments)
+  end
+
   def create
-    @post = Post.find(params[:post_id])
-    @comment = @post.comments.new(params.require(:comment).permit(:text))
-    @comment.user_id = current_user.id
-
-    respond_to do |format|
-      format.html do
-        if @comment.save
-          redirect_to user_post_path(current_user, @post), notice: 'Comment created successfully!'
-        else
-          # error message
-          flash.now[:error] = 'Error: Post could not be created'
-          # render new
-          render :new, locals: { comment: }
-        end
-      end
-    end
+    @comment = @post.comments.create!(comment_params)
+    json_response(@comment, :created)
+
+    # @post = Post.find(params[:post_id])
+    # @comment = @post.comments.new(params.require(:comment).permit(:text))
```

Comment:
Optional
Please remove any commented out code for a more optimized and cleaner codebase